### PR TITLE
Refactor to use API in visual modes

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Refactor panels to use API instead of `rz_core_cmd*()` calls. Also partially converted `aaa` command to the API.

**Test plan**

All green.

**Closing issues**

Partially addresses #383

### CODE REVIEW

File: librz/core/panels.c [hunk 1]
```diff
@@ -4065,7 +4065,7 @@ void __hudstuff(RzCore *core) {
 
 void __esil_init(RzCore *core) {
 	rz_core_analysis_esil_init_mem(core, NULL, UT64_MAX, UT32_MAX);
-	rz_core_cmd(core, "aeip", 0);
+	rz_core_analysis_set_reg(core, "PC", core->offset);
```
Comment (ret2libc) [inline librz/core/panels.c]: I think it makes sense to have a `rz_core_analysis_esil_init_regs` function in canalysis.c. We may have to do additional configuration steps on `aeip` than just PC=offset?

File: librz/core/cmd_analysis.c [hunk 2]
```diff
@@ -8724,159 +8706,11 @@ static int cmd_analysis_all(RzCore *core, const char *input) {
...
-				}
-				rz_core_task_yield(&core->tasks);
-				oldstr = rz_print_rowlog(core->print, "Check for vtables");
-				rz_core_cmd0(core, "avrr");
-				rz_print_rowlog_done(core->print, oldstr);
-				rz_core_task_yield(&core->tasks);
-				rz_config_set_i(core->config, "analysis.calls", c);
-				rz_core_task_yield(&core->tasks);
-				if (rz_cons_is_breaked()) {
+				bool experimental = true ? (input[1] == 'a') : false;
```
Comment (ret2libc) [inline librz/core/cmd_analysis.c]: ?????

File: librz/core/canalysis.c [hunk 3]
```diff
@@ -6460,3 +6460,180 @@ RZ_IPI bool rz_core_analysis_var_rename(RzCore *core, const char *name, const ch
...
+	ut64 t = rz_num_math(core->num, "$S");
+	rz_core_seek(core, t, true);
+	if (rz_cons_is_breaked()) {
+		return false;
+	}
+
+	oldstr = rz_print_rowlog(core->print, "Analyze function calls (aac)");
+	(void)rz_cmd_analysis_calls(core, "", false, false); // "aac"
+	rz_core_seek(core, curseek, true);
+	// oldstr = rz_print_rowlog (core->print, "Analyze data refs as code (LEA)");
```
Comment (ret2libc) [inline librz/core/canalysis.c]: just delete

File: librz/core/canalysis.c [hunk 4]
```diff
@@ -6460,3 +6460,180 @@ RZ_IPI bool rz_core_analysis_var_rename(RzCore *core, const char *name, const ch
...
+	const char *arch = rz_config_get(core->config, "asm.arch");
+	if (!strstr(arch, "ppc") && !strstr(arch, "arm") && !strstr(arch, "x86")) {
+		return false;
+	}
+	RzBinObject *bo = rz_bin_cur_object(core->bin);
+	rz_return_val_if_fail(!bo || (bo->plugin && bo->plugin->name), false);
+	return bo ? strstr(bo->plugin->name, "mach") : false;
+}
+
+// Returns true if all steps were finished and false if it was interrupted
```
Comment (ret2libc) [inline librz/core/canalysis.c]: Use doxygen-style doc if any

File: librz/core/cmd_analysis.c (hunk 2 above)
Comment (XVilka) [reply librz/core/cmd_analysis.c]: I translated the logic of `aaaa` detection further in the function.

File: librz/core/panels.c (hunk 1 above)
Comment (XVilka) [reply librz/core/panels.c]: Done.

File: librz/core/cmd_analysis.c (hunk 2 above)
Comment (ret2libc) [reply librz/core/cmd_analysis.c]: I think you meant
```suggestion
				bool experimental = input[1] == 'a';
```

**Review by ret2libc — APPROVED** [review]

**Pull request merged**